producer/writer/message_writer.go

```diff
@@ -154,28 +158,30 @@ func newMessageWriter(
 	if opts == nil {
 		opts = NewOptions()
 	}
+	nowFn := time.Now
```

Comment:
Pass this in through clock options in the options?

Reply:
I feel since this is only used in test, I'd rather just override it in tests